pkg/controller/job/pod_failure_policy.go

```diff
@@ -70,25 +70,15 @@ func matchOnExitCodes(podStatus *v1.PodStatus, requirement *batch.PodFailurePoli
 	return getMatchingContainerFromList(podStatus.InitContainerStatuses, requirement)
 }
 
-func matchOnPodConditions(podStatus *v1.PodStatus, requirement *batch.PodFailurePolicyOnPodConditionsRequirement) *v1.PodCondition {
+func matchOnPodConditions(podStatus *v1.PodStatus, requirement []batch.PodFailurePolicyOnPodConditionsPattern) *v1.PodCondition {
 	for _, podCondition := range podStatus.Conditions {
-		if isOnPodConditionsOperatorMatching(podCondition.Type, requirement.Operator, requirement.Values) {
-			return &podCondition
-		}
-	}
-	return nil
-}
-
-func isOnPodConditionsOperatorMatching(conditionType v1.PodConditionType, operator batch.PodFailurePolicyOnPodConditionsOperator, values []v1.PodConditionType) bool {
-	switch operator {
-	case batch.PodFailurePolicyOnPodConditionsOpIn:
-		for _, value := range values {
-			if value == conditionType {
-				return true
+		for _, pattern := range requirement {
+			if podCondition.Type == pattern.Type {
```

Comment:
match the status to True.

And add a unit test for status=False that shouldn't match

Reply:
done